benchmarks/README.md

```diff
@@ -0,0 +1,88 @@
+# Benchmarks
+
+## DrWatson Initialization
+
+This code base is using the [Julia Language](https://julialang.org/) and
+[DrWatson](https://juliadynamics.github.io/DrWatson.jl/stable/)
+to make a reproducible scientific project named
+> benchmarks
+
+To (locally) reproduce this project, do the following:
+
+0. Download this code base. Notice that raw data are typically not included in the
+   git-history and may need to be downloaded independently.
+1. Open a Julia console and do:
+   ```
+   julia> using Pkg
+   julia> Pkg.add("DrWatson") # install globally, for using `quickactivate`
+   julia> Pkg.activate("path/to/this/project")
+   julia> Pkg.instantiate()
+   ```
+
+This will install all necessary packages for you to be able to run the scripts and
+everything should work out of the box, including correctly finding local paths.
+
+You may notice that most scripts start with the commands:
+```julia
+using DrWatson
+@quickactivate :benchmarks
+```
+which auto-activate the project, enable local path handling from DrWatson and provide several helper functions.
+
+## Establishing Configurations
+
+To establish a set of configurations for a simulation, you should use ```config_generate.jl```, located in the ```scripts``` directory. This will read either the included ```main_config.toml``` or a user provided file, when the file name is provided as a command line argument, and distribute the configuration information to multiple tasks. 
+
+To use a custom TOML, please follow all the steps below:
+
+0. The user provided TOML must located in the ```src``` directory.
+1. For a given simulation (heat, brusselator, etc.) and intended architeture (cpu, cuda), include a group in the TOML as `[sim_name.arch]`, where `sim_name` and `arch` are respective names from the previous groups.
+2. Under a group, list all the parameters desired. This should be structured as either `param_name = [val1, val2, val3...]` or `param_name = val`.
+3. Highly recommended parameters to include are ```code_target```, which takes either ```CPUTarget``` or ```CUDATarget```.
+
+```config_generate.jl``` will generate the required directories automatically. If you were to include ```[heat.cpu]``` in your TOML, the resulting configuration would be in ```src/heat/heat_cpu.toml```.
+
+Please view ```main_config.toml``` as a guiding example on how to craft your own TOML. 
+
+**Warning**: ```config_generate.jl``` is not called automatically so it is up to you to run the script before launching benchmarks. 
```

Comment:
If you don't call this the error message is quite cryptic. maybe the main.jl script should check that the configs exist and if not, then automatically call `config_generate` for you?

Reply:
Yeah that was the cryptic error. I feel like a makefile approach of conditional compilation is such a good fit for what you are doing. If the individual configs don't exists, then you need to regenerate them. But I think we can get rid of that step entirely by a later comment